### ISSUE TITLE
`test_rechunk_in_memory[p2p]` raises `RuntimeError: Identity cache already includes the item.`

### ISSUE THREAD

**Issue by hendrikmakait** [coiled/benchmarks#789]
```
_________________________ test_rechunk_in_memory[p2p] __________________________
[gw0] linux -- Python 3.9.16 /usr/share/miniconda3/envs/test/bin/python

small_client = <Client: 'tls://10.0.13.228:8786' processes=10 threads=20, memory=71.68 GiB>
configure_rechunking = None

    @run_up_to_nthreads("small_cluster", 50, reason="fixed dataset")
    def test_rechunk_in_memory(small_client, configure_rechunking):
        x = da.random.random((50000, 50000))
>       x.rechunk((50000, 20)).rechunk((20, 50000)).sum().compute()

tests/benchmarks/test_array.py:235: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/share/miniconda3/envs/test/lib/python3.9/site-packages/dask/base.py:314: in compute
    (result,) = compute(self, traverse=False, **kwargs)
/usr/share/miniconda3/envs/test/lib/python3.9/site-packages/dask/base.py:599: in compute
    results = schedule(dsk, keys, **kwargs)
/usr/share/miniconda3/envs/test/lib/python3.9/site-packages/distributed/client.py:3213: in get
    results = self.gather(packed, asynchronous=asynchronous, direct=direct)
/usr/share/miniconda3/envs/test/lib/python3.9/site-packages/distributed/client.py:2348: in gather
    return self.sync(
/usr/share/miniconda3/envs/test/lib/python3.9/site-packages/distributed/utils.py:349: in sync
    return sync(
/usr/share/miniconda3/envs/test/lib/python3.9/site-packages/distributed/utils.py:416: in sync
    raise exc.with_traceback(tb)
/usr/share/miniconda3/envs/test/lib/python3.9/site-packages/distributed/utils.py:389: in f
    result = yield future
/usr/share/miniconda3/envs/test/lib/python3.9/site-packages/tornado/gen.py:769: in run
    value = future.result()
/usr/share/miniconda3/envs/test/lib/python3.9/site-packages/distributed/client.py:2211: in _gather
    raise exception.with_traceback(traceback)
/opt/coiled/env/lib/python3.9/site-packages/dask/array/random.py:816: in _apply_random
    ???
mtrand.pyx:185: in numpy.random.mtrand.RandomState.__init__
    ???
_mt19937.pyx:166: in numpy.random._mt19937.MT19937._legacy_seeding
    ???
_mt19937.pyx:189: in numpy.random._mt19937.MT19937._legacy_seeding
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   RuntimeError: Identity cache already includes the item.

/opt/coiled/env/lib/python3.9/site-packages/numpy/core/_methods.py:57: RuntimeError
```

* #784 (https://github.com/coiled/benchmarks/actions/runs/4673168030/jobs/8276101113#step:6:445)
* #786 (https://github.com/coiled/benchmarks/actions/runs/4682206701/jobs/8295901271#step:6:446)

**Comment by hendrikmakait** [comment]
@jrbourbeau: Have you seen this error message before? It looks like it's caused by `array.random` rather than P2P rechunking.

**Comment by hendrikmakait** [comment]
Thanks for pointing to the upstream PR and fixing, @jrbourbeau.

Closed via #790.
